### DISCUSSION TITLE
Which proxy service works best for this api / youtube.com?

### DISCUSSION THREAD

**Discussion by ne1baf** [jdepoix/youtube-transcript-api#335] [General]
I'm a brightdata customer but I tried both residential and data center proxies, they both lead to 403. I tried scraperAPI and rotating proxies and after 200 attemps with residentatial proxies, still getting 403... ouch

Anyone has recommendations?

**Comment by achillion** [comment]
Anyone can answer this question please?
OP, do you have the answer?

I know some proxy services block youtube, and you need to pass through some hoops with KYC and stuff, but appart from that, i would love to know what people are using?

**Reply by ne1baf** [reply]
Hi, I used webshare's static residential proxies. You pay $6/mo and you get 20 proxies. Out of those, I usually got 10-15 working ones for YT. There is no KYC from what I remember. You just must make sure to not burn them because you cannot ask for a new one. Hope it helps.

```
transcript = YouTubeTranscriptApi.get_transcript(video_id, languages=[lang], proxies=proxy)
```

**Reply by achillion** [reply]
Thank you for that, that's also what i ended up using. So far it's enough for my usage. Thanks

**Reply by jdepoix** [reply]
See my answer below! 😊

**Comment by jdepoix** [comment]
Hi @ne1baf!

Since I have been getting a lot of messages about which provider of residential proxies I would recommend, I have tested a lot of different providers throughout the last months to be able to give a well-founded recommendation. While there's certainly different viable options, I have found [Webshare's Residential Proxies](https://www.webshare.io/?referral_code=w0xno53eb50g) to work most reliably! Therefore I have added a [Webshare integration](https://github.com/jdepoix/youtube-transcript-api?tab=readme-ov-file#using-webshare) in version [v1.0.0](https://github.com/jdepoix/youtube-transcript-api/releases/tag/v1.0.0) , to make setting it up as easy as possible:
```python
from youtube_transcript_api import YouTubeTranscriptApi
from youtube_transcript_api.proxies import WebshareProxyConfig

ytt_api = YouTubeTranscriptApi(
	# username and password can be retrieved from https://dashboard.webshare.io/proxy/settings
	proxy_config=WebshareProxyConfig(
		proxy_username="<proxy-username>",
		proxy_password="<proxy-password>",
	)
)

# all requests done by ytt_api will now be proxied through Webshare
ytt_api.fetch(video_id)
```
Using the `WebshareProxyConfig` will default to using rotating residential proxies and requires no further configuration! 🪄

You can find more information on this in the new [Working around IP bans](https://github.com/jdepoix/youtube-transcript-api?tab=readme-ov-file#working-around-ip-bans-requestblocked-or-ipblocked-exception) section of the README!

**Reply by achillion** [reply]
Any breaking change with this 1.0 update?

**Reply by jdepoix** [reply]
If you want to use the new `.fetch` and `.list` methods, yes! But the old static methods (`get_transcript`, `get_transcripts` and `list_transcripts`) can still be used, to make the transition to the new version as easy as possible. However, they have been deprecated and will be removed in a future version!

You'll find more details on these changes in the [v1.0.0 release notes](https://github.com/jdepoix/youtube-transcript-api/releases/tag/v1.0.0)!

**Comment by timfong888** [comment]
How can I tell if the issues I am having are with the proxy I am using. Even after the release updating how captions work, I am unable to get it to work on a cloud server. 

I do not get RequestBlocked or IpBlocked exceptions.

**Reply by jdepoix** [reply]
Please make sure you're using the latest version. If the issue remains open an issue and please fill out the issue template.

**Comment by DavidNaak** [comment]
@jdepoix I'm currently using Webshare with rotating proxies and it works, but I'm seeing latency between 3–12 seconds, often 5 seconds or more. That’s too high for my web app. Are you experiencing similar delays, or could I be doing something wrong?

**Reply by jdepoix** [reply]
I already answered your question [in this comment](https://github.com/jdepoix/youtube-transcript-api/issues/472#issuecomment-3103023893)

**Reply by chaitanyame** [reply]
@jdepoix  is there anyway transcript size can be reduced, during the proxy usage it is using lot of bandwidth  and rotating proxy charge per gb.

**Reply by jdepoix** [reply]
@chaitanyame unfortunately no. We need to fetch the HTML for the entire webview of the video we want to retrieve the transcripts for, to be able to extract the required information, as the keys get generated by YouTube per request. Fetching the HTML makes up the biggest chunk of the required bandwidth, but there is no way around that.
